From: Shaohua Li <shli@fb.com>
Date: Thu, 8 Sep 2016 10:49:06 -0700
Subject: raid5: allow arbitrary max_hw_sectors
References: bsc#1003941,FATE#321732
Git-commit: 1dffddddd8315863f1a6d79c512b737864ef6a1a
Patch-Mainline: v4.9-rc1

raid5 will split bio to proper size internally, there is no point to use
underlayer disk's max_hw_sectors. In my qemu system, without the change,
the raid5 only receives 128k size bio, which reduces the chance of bio
merge sending to underlayer disks.

Signed-off-by: Shaohua Li <shli@fb.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/raid5.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/md/raid5.c b/drivers/md/raid5.c
index 48f50d7..29414a1 100644
--- a/drivers/md/raid5.c
+++ b/drivers/md/raid5.c
@@ -7075,6 +7075,8 @@ static int raid5_run(struct mddev *mddev)
 		else
 			queue_flag_clear_unlocked(QUEUE_FLAG_DISCARD,
 						mddev->queue);
+
+		blk_queue_max_hw_sectors(mddev->queue, UINT_MAX);
 	}
 
 	if (journal_dev) {
-- 
1.8.5.6

